sdk/python/foundation-models/system/finetune/Llama-notebooks/distillation/distillation_math.ipynb

```diff
@@ -0,0 +1,599 @@
+   "source": [
+    "## Prepare data inputs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data = None\n",
+    "train_data_name = \"math_train_multi_arith\"\n",
+    "\n",
+    "train_data = ml_client.data.create_or_update(\n",
+    "    Data(\n",
+    "        path=train_data_path,\n",
+    "        type=AssetTypes.URI_FILE,\n",
+    "        description=\"Training dataset\",\n",
+    "        name=train_data_name,\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "train_data_asset_id = f\"azureml://locations/{ai_project.location}/workspaces/{ai_project._workspace_id}/data/{train_data.name}/versions/{train_data.version}\"\n",
+    "train_data_asset_id"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "valid_data = None\n",
+    "valid_data_name = \"math_valid_multi_arith\"\n",
+    "\n",
+    "valid_data = ml_client.data.create_or_update(\n",
+    "    Data(\n",
+    "        path=valid_data_path,\n",
+    "        type=AssetTypes.URI_FILE,\n",
+    "        description=\"validation dataset\",\n",
+    "        name=valid_data_name,\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "valid_data_asset_id = f\"azureml://locations/{ai_project.location}/workspaces/{ai_project._workspace_id}/data/{valid_data.name}/versions/{valid_data.version}\"\n",
+    "valid_data_asset_id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Distillation strategy settings\n",
+    "\n",
+    "We provide the option to leverage Chain of Thought (CoT) reasoning for distillation. CoT leverages step by step reasoning ability of the teacher model to generate more accurate labels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ENABLE_CHAIN_OF_THOUGHT = \"true\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Configure distillation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mlclient_azureml = MLClient(credential, registry_name=\"azureml\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "distillation_pipeline_name = \"oss_distillation_pipeline\"\n",
+    "distillation_pipeline_component = mlclient_azureml.components.get(\n",
+    "    name=distillation_pipeline_name\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Select Task Type\n",
+    "\n",
+    "For Math distillation, there are two task types. `MATH` and `MATH_MCQ` are both math task types. \n",
+    "\n",
+    "For math datasets, such as MultiArith (current dataset), where the answer is numeric, select `MATH` as the `data_generation_task_type`. \n",
```

Comment:
Have we added this to the validation and distillation component in azureml-assets ? We should be careful about the order of submission of the CLs.

Reply:
Yes a PR is in in review that includes the MATH task in the validation and data-gen components. That will go in first before merging this PR (also MATH_MCQ is being removed as a task and has been updated in this PR